cmd/thanos/bucket.go

```diff
@@ -213,4 +217,191 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
 			return printBlock(id)
 		})
 	}
+
+	inspect := cmd.Command("inspect", "inspect all blocks in the bucket")
+	selector := inspect.Flag("selector ", "Selects blocks based on label, e.g. -l key1=value1,key2=value2. All key value pairs must match.").Short('l').
+		Default("").String()
+	sortBy := inspect.Flag("sort-by", "Sort by columns.").
+		Default("FROM,UNTIL").String()
+
+	m[name+" inspect"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
+
+		// Parse selector.
+		selectorMap := make(map[string]string)
+		selectorsSplit := strings.Split(*selector, ",")
+		for _, s := range selectorsSplit {
+			selectorSplit := strings.Split(s, "=")
+			if len(selectorSplit) == 2 {
+				selectorMap[selectorSplit[0]] = selectorSplit[1]
+			}
+		}
+
+		bucketConfig, err := objStoreConfig.Content()
+		if err != nil {
+			return err
+		}
+
+		bkt, err := client.NewBucket(logger, bucketConfig, reg, name)
+		if err != nil {
+			return err
+		}
+
+		// Dummy actor to immediately kill the group after the run function returns.
+		g.Add(func() error { return nil }, func(error) {})
+
+		defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+
+		// Getting Metas.
+		var blockMetas []*block.Meta
+		err = bkt.Iter(ctx, "", func(name string) error {
```

Comment:
You can inline err check:

` if err := bkt.Iter... ; err != nil {`

Reply:
done